nixos/common/flake.nix

```diff
@@ -42,6 +57,9 @@ in
     };
   };
   config = lib.mkIf (cfg.flake != null) {
+    nix.settings.nix-path = lib.mkIf (cfg.flake.inputs ? nixpkgs) "nixpkgs=${cfg.flake.inputs.nixpkgs}";
```

Comment:
This can be a bit wonky in nix versions < 2.24: https://github.com/NixOS/nix/issues/9574#issue-2033997689